liblustreapi/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use std::convert::From;
+use std::ffi::{CStr, CString};
+use std::fmt;
+use std::io;
+extern crate liblustreapi_sys as sys;
+static PATH_BYTES: usize = 4096;
+
+#[derive(Copy, Clone)]
+pub struct Fid {
+    pub seq: u64,
+    pub oid: u32,
+    pub ver: u32,
+}
+impl fmt::Display for Fid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "[0x{:x}:0x{:x}:0x{:x}]", self.seq, self.oid, self.ver)
+    }
+}
+// @@ There's got to be a better way to do this!
+fn num2u32(num: &str) -> u32 {
+    if num.starts_with("0x") {
+        u32::from_str_radix(num.trim_start_matches("0x"), 16).unwrap()
+    } else if num.starts_with("0X") {
+        u32::from_str_radix(num.trim_start_matches("0X"), 16).unwrap()
+    } else {
+        num.parse().unwrap()
+    }
+}
+fn num2u64(num: &str) -> u64 {
+    if num.starts_with("0x") {
+        u64::from_str_radix(num.trim_start_matches("0x"), 16).unwrap()
+    } else if num.starts_with("0X") {
+        u64::from_str_radix(num.trim_start_matches("0X"), 16).unwrap()
+    } else {
+        num.parse().unwrap()
+    }
+}
+impl From<String> for Fid {
+    fn from(fidstr: String) -> Self {
+        let fidstr = fidstr.trim_matches(|c| c == '[' || c == ']');
+        let arr: Vec<&str> = fidstr.split(':').collect();
+        Fid {
+            seq: num2u64(arr[0]),
+            oid: num2u32(arr[1]),
+            ver: num2u32(arr[2]),
+        }
+    }
+}
+impl From<[u8; 40usize]> for Fid {
```

Comment:
This will need to be updated as well, it contains an `.unwrap()` still.

```rust
impl From<[u8; 40usize]> for Fid {
    fn from(fidstr: [u8; 40usize]) -> Self {
        String::from_utf8_lossy(&fidstr)
            .into_owned()
            .parse::<Fid>()
            .unwrap()
    }
}
```